src/VisualStudio/CSharp/Test/DocumentOutline/DocumentOutlineTests.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServices;
+using Roslyn.Utilities;
+using Xunit;
+using Range = Microsoft.VisualStudio.LanguageServer.Protocol.Range;
+
+namespace Roslyn.VisualStudio.CSharp.UnitTests.DocumentOutline
+{
+    public class DocumentOutlineTests
+    {
+        private const int NumParentNodes = 3;
+        private const int NumChildNodes = 3;
+
+        // Mocks a nested DocumentSymbol[] returned from an LSP document symbol request
+        private static DocumentSymbol[] GetDocumentSymbols()
+        {
+            var documentSymbols = Array.Empty<DocumentSymbol>();
```

Comment:
Given we know the size this is going to be, just allocate rather than resizing each time?